plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/BulkIndexByScrollResponse.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.reindex;
+
+import static java.lang.Math.min;
+import static java.util.Collections.unmodifiableList;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.BATCHES;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.FAILURES;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.NOOPS;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.TOOK;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.UPDATED;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.VERSION_CONFLICTS;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.bulk.BulkItemResponse.Failure;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+
+/**
+ * Response used for actions that index many documents using a scroll request.
+ */
+public class BulkIndexByScrollResponse extends ActionResponse implements ToXContent {
+    private long took;
+    private long updated;
+    private int batches;
+    private long versionConflicts;
+    private long noops;
+    private List<Failure> failures;
+
+    public BulkIndexByScrollResponse() {
+    }
+
+    public BulkIndexByScrollResponse(long took, long updated, int batches, long versionConflicts, long noops, List<Failure> failures) {
+        this.took = took;
+        this.updated = updated;
+        this.batches = batches;
+        this.versionConflicts = versionConflicts;
+        this.noops = noops;
+        this.failures = unmodifiableList(failures);
+    }
+
+    public long took() {
+        return took;
+    }
+
+    public long updated() {
+        return updated;
+    }
+
+    public int batches() {
```

Comment:
For the response objects, can we have proper getters? Would be more in line with the other Response objects in the codebase.